perf/pgbench.py

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2023 IBM
+# Author: Gautam Menghani <gautam@linux.ibm.com>
+
+import os
+
+from avocado import Test
+from avocado.utils import process, cpu
+from avocado.utils.service import ServiceManager
+from avocado.utils.software_manager.manager import SoftwareManager
+
+
+class Pgbench(Test):
+
+    """
+    This module will run the pgbench benchmark. Pgbench is a benchmark for
+    measuring the throughput achieved by executing database transactions on
+    PostgresSQL database.
+    """
+    sm = SoftwareManager()
+    scaling_factor = worker_threads = db_clients = protocol = \
+        benchmark_duration = transaction_count = 0
+
+    def run_cmd(self, cmd, ignore_failure):
+        command = process.SubProcess(cmd)
+        command.run()
+
+        if command.result.exit_status and ignore_failure:
+            self.log.info(f"Command '{cmd}' returned '{command.result.exit_status}' "
+                          f"with the text '{command.result.stderr_text}'")
+            return [command.result.exit_status, command.result.stderr_text]
+        elif command.result.exit_status:
+            self.cancel(f"Command '{cmd}' returned '{command.result.exit_status}' "
+                        f"with the text '{command.result.stderr_text}'")
+        else:
+            return [command.result.exit_status, command.result.stdout_text]
+
+    def get_pkg_manager(self):
+        for pkg_mgr in ['apt', 'dnf', 'yum']:
+            if self.sm.check_installed(pkg_mgr):
+                return pkg_mgr
+        return None
+
+    def install_packages(self, pkg_mgr):
+        packages = ["postgresql", "postgresql-contrib"]
+        for package in packages:
+            self.sm.install(package)
+
+        # rhel needs an extra pkg
+        if pkg_mgr in ['dnf', 'yum']:
+            self.sm.install('postgresql-server')
+
+            # restart postgres service
+            ManagerService = ServiceManager()
+            ManagerService.restart("postgresql")
+
+    def setUp(self):
+        '''
+        Install Pgbench
+        '''
+        # check root privileges ($USER should be able to run commands
+        # as another user without requiring a password)
+        if os.geteuid() != 0:
+            self.cancel("This script requires root privileges, "
+                        "Please try again with 'sudo' or as 'root'")
+
+        # check if pgbench is installed
+        if (self.sm.check_installed("pgbench")):
```

Comment:
shouldnt you restart pgbench or ensure that its running if the pgbench package is already installed